spyder/workers/updates.py

```diff
@@ -225,16 +227,38 @@ def _progress_reporter(self, block_number, read_size, total_size):
     def _download_installer(self):
         """Donwload latest Spyder standalone installer executable."""
         tmpdir = tempfile.gettempdir()
-        is_full_installer = (is_module_installed('numpy') or
-                             is_module_installed('pandas'))
-        if os.name == 'nt':
-            name = 'Spyder_64bit_{}.exe'.format('full' if is_full_installer
-                                                else 'lite')
+
+        # If running on macOS under Rosetta, platform.machine will not be
+        # correct so check for ARM64 in version instead. All other
+        # platforms will be x86_64.
+        mach = "arm64" if "ARM64" in platform.version() else "x86_64"
```

Comment:
And also this if users installed our Intel installer in an M1 machine, right?

Reply:
We could consider it. Currently on master we already detect arm architecture using the standard `platform.machine()`. I think it highly unlikely that a user will install an x86_64 version of Spyder 6.0 on M1 (though, I suppose it is theoretically possible). The above is certainly more robust. If necessary, I can update master in a separate PR.

Reply:
>  I think it highly unlikely that a user will install an x86_64 version of Spyder 6.0 on M1 (though, I suppose it is theoretically possible). The above is certainly more robust. If necessary, I can update master in a separate PR.

I think it's better to cover that case for newbies that are not aware of the difference between Intel and M1 and could grab any installer that has `macOS` on its name.